### PR TITLE
Completed Assignment 4 tasks and modified DatasetDescriptions of Assignement1

### PR DESCRIPTION
**Pull request by Meril99**
As you told me per message I re-modified Dataset of Assignement 1 because my previous changes weren' taken into account since I didn't sue Git in roder to change it. Furthermore I created my repo for assignement 4 with Meril99 because I still don't have a StudentID

### CODE REVIEW

**Comment by dgarijo** [comment]
@Meril99,
you are adding many changes into the dataset descriptions file. Please add only YOUR line, do not modify the rest of the file

File: Assignment4/MiangouilaMeril99/task06.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the email address, fullName, given and family names**"""
+
+jane_smith = ns.JaneSmith
+g.add((jane_smith, vcard.EMAIL, Literal("jane.smith@example.com")))
+g.add((jane_smith, vcard.Given, Literal("Jane")))
+g.add((jane_smith, vcard.Family, Literal("Smith")))
```
Comment (dgarijo) [inline Assignment4/MiangouilaMeril99/task06.py:68]: where is Jane`s formatted name?

File: Assignment4/MiangouilaMeril99/task06.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+john_smith = ns.JohnSmith
+upm = ns.UPM
+g.add((upm, RDF.type, ns.University))
+g.add((john_smith, vcard.ORG, upm))
```
Comment (dgarijo) [inline Assignment4/MiangouilaMeril99/task06.py:79]: I do not see ORG in vcard. You should find another property or define your own.

File: Assignment4/MiangouilaMeril99/task07.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+# Import necessary libraries
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+query_text = """
+    PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+    PREFIX ns: <http://somewhere#>
+    SELECT ?subclass
+    WHERE {
+        ?subclass rdfs:subClassOf ns:LivingThing .
```
Comment (dgarijo) [inline Assignment4/MiangouilaMeril99/task07.py:34]: missing all subclasses of subclasses of living thing

File: Assignment4/MiangouilaMeril99/task07.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+"""
+
+query_text = """
+    PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+    PREFIX ns: <http://somewhere#>
+    SELECT distinct ?individual ?property ?value
+    WHERE {
+        ?individual rdf:type ?class .
+        ?individual ?property ?value .
+        FILTER (?class = ns:Person || ?class = ns:Animal) .
```
Comment (dgarijo) [inline Assignment4/MiangouilaMeril99/task07.py:70]: Use UNION instead, as this filter is quite inefficient.

File: Assignment4/MiangouilaMeril99/task07.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+query_text = """
+    PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+    SELECT ?entity
+    WHERE {
+        ?entity foaf:knows ?known1 .
+        ?entity foaf:knows ?known2 .
+        FILTER (?known1 != ?known2)
+    }
+"""
+q5 = prepareQuery(query_text)
```
Comment (dgarijo) [inline Assignment4/MiangouilaMeril99/task07.py:106]: Missing all exercises in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Changes are needed. Please have a look

File: Assignment4/MiangouilaMeril99/task07.py (hunk 5 above)
Comment (Meril99) [reply Assignment4/MiangouilaMeril99/task07.py:106]: what do you mean ? I used rdflib as shown in the class

File: Assignment4/MiangouilaMeril99/task06.py (hunk 1 above)
Comment (Meril99) [reply Assignment4/MiangouilaMeril99/task06.py:68]: It was not asked for this question. I had answered it in the queston just before

File: Assignment4/MiangouilaMeril99/task07.py (hunk 5 above)
Comment (dgarijo) [reply Assignment4/MiangouilaMeril99/task07.py:106]: You are doing a sparql query for all of them.
Rdflib does iterates through the graph

**Pull request closed without merging**